### PR TITLE
cryptonite-app

### PR DESCRIPTION
**Pull request by Aitor1111**

### CODE REVIEW

File: staff/groups/team-jalapeno/cryptonite/components/App.jsx [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+        setView('login')
+    }
+
+    const handleLoginSubmit = (event) => {
+        event.preventDefault()
+
+        const email = event.target.email.value
+        const password = event.target.password.value
+
+        authenticateUser(email, password, (error, token) => {
```
Comment (lichab) [inline staff/groups/team-jalapeno/cryptonite/components/App.jsx:27]: This function shold be wrapped in a try catch in order to handle the synchronous errors. The error you are taking into account with the callback is only for asynchronous operations.  This is the same in the whole code so be careful with that.

File: staff/groups/team-jalapeno/cryptonite/logic/toggle-favorite.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
+function toggleFavorite(token, id, callback){   // logica para pasar un coin a favorite
+    String.validate.notVoid(token)
+    String.isString(id)
+    Function.validate(callback)
+    
+    retrieveUser(token, (error, user)=> {
```
Comment (lichab) [inline staff/groups/team-jalapeno/cryptonite/logic/toggle-favorite.js:6]: You can't use this logic function inside of another one, this does not follow good practices. As done during the bootcamp you should make this with a call()

File: staff/groups/team-jalapeno/cryptonite/components/App.jsx (hunk 1 above)
Comment (Aitor1111) [reply staff/groups/team-jalapeno/cryptonite/components/App.jsx:33]: done!

File: staff/groups/team-jalapeno/cryptonite/logic/toggle-favorite.js (hunk 2 above)
Comment (Aitor1111) [reply staff/groups/team-jalapeno/cryptonite/logic/toggle-favorite.js]: done!

File: staff/groups/team-jalapeno/cryptonite/logic/!!!get-portfolio-data.js [hunk 3]
```diff
@@ -0,0 +1,31 @@
+retrieveUser(sessionStorage.token, (userError, user) => {
```
[the remark this answers is not in the document]
Comment (Aitor1111) [reply staff/groups/team-jalapeno/cryptonite/logic/!!!get-portfolio-data.js]: Sorry, that logic file was temporary, finally deleted.

File: staff/groups/team-jalapeno/cryptonite/logic/add-portfolio-crypto.js [hunk 4]
```diff
@@ -0,0 +1,49 @@
+function addPortfolioCrypto(token, portfolioInfo, callback) {
+
+    String.validate.notVoid(token)
+    Function.validate(callback)
+
+
+    let url = 'https://skylabcoders.herokuapp.com/api/v2/users/'
```
[the remark this answers is not in the document]
Comment (Aitor1111) [reply staff/groups/team-jalapeno/cryptonite/logic/add-portfolio-crypto.js]: Done thx

File: staff/groups/team-jalapeno/cryptonite/logic/delete-from-portfolio.js [hunk 5]
```diff
@@ -0,0 +1,38 @@
+
+function deletePortfolioCrypto(token, cryptoId, callback) {
+    String.validate.notVoid(token)
+    String.validate.notVoid(cryptoId)
+    Function.validate(callback)
+
+    let url = 'https://skylabcoders.herokuapp.com/api/v2/users/'
```
[the remark this answers is not in the document]
Comment (Aitor1111) [reply staff/groups/team-jalapeno/cryptonite/logic/delete-from-portfolio.js]: Done

File: staff/groups/team-jalapeno/cryptonite/components/PortfolioPage.jsx [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+                    setWallet(total)
+                    setCoinCryptos(_data)
+                }, ids)
+            }
+        })
+
+    }, [])
+
+    const subTotal = (quantity, priceUsd) => {
+        let result = Number(quantity) * Number(priceUsd)
```
[the remark this answers is not in the document]
Comment (Aitor1111) [reply staff/groups/team-jalapeno/cryptonite/components/PortfolioPage.jsx]: Solved!

File: staff/groups/team-jalapeno/cryptonite/components/CryptosListPage.jsx [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+
+          setWallet(total)
+        }, ids)
+      }
+    })
+
+  }, [])
+
+  const subTotal = (quantity, priceUsd) => {
+    let result = Number(quantity) * Number(priceUsd)
```
[the remark this answers is not in the document]
Comment (Aitor1111) [reply staff/groups/team-jalapeno/cryptonite/components/CryptosListPage.jsx]: Solved

File: staff/groups/team-jalapeno/cryptonite/components/CoinPage.jsx [hunk 8]
```diff
@@ -0,0 +1,134 @@
...
+    const [inPortfolio, setInPortfolio] = useState(false)
+
+
+    useEffect(() => {
+        let coinName = window.location.hash
+        if (!coinName) return
+
+        coinName = coinName.split('/')[1]
+
+        retrieveCrypto(coinName, (_error, _crypto) => {
```
[the remark this answers is not in the document]
Comment (Aitor1111) [reply staff/groups/team-jalapeno/cryptonite/components/CoinPage.jsx]: Done trycatch!

**Comment by Tanane** [comment]
Thank you Lisandro for your feedback, I have resolved everything that you've pointed out.

**Pull request merged**